editor.planx.uk/src/pages/FlowEditor/lib/analytics/utils.ts

```diff
@@ -131,22 +133,34 @@ export function getAnswers(
   return answers;
 }
 
+const sanitiseAllowListValues = ([ key, value ]: [key: string, value: unknown]): [string, unknown] => {
+  if (!isObject(value)) return [key, value];
+
+  // Strip out empty values - we do not need to store these as allow list answers
+  const sanitisedObject = Object.fromEntries(
+    Object.entries(value).filter(([_key, value]) => Boolean(value))
+  );
+  if (isEmpty(sanitisedObject)) return [key, undefined];
+
+  return [key, sanitisedObject];
+}
+
 /**
  * Extract allowlist answers from breadcrumb data
  * e.g. data set automatically by components such as DrawBoundary
  */
 export function getData(
   breadcrumb: Store.UserData,
-): Partial<Record<AllowListKey, any>> | undefined {
+): Partial<Record<AllowListKey, unknown>> | undefined {
   const dataSetByNode = breadcrumb.data;
   if (!dataSetByNode) return;
 
   const filteredEntries = Object.entries(dataSetByNode)
+    .map(sanitiseAllowListValues)
     .filter(([key, value]) => isAllowListKey(key) && Boolean(value))
```

Comment:
🐛 Bug! Here's the reason we're saving empty feedback values as allow list answers.

`Boolean(value)` is too simplistic when checking that allow list values are populated. In our feedback example the current object is always truthy (which passes this check).

```ts
{ 
  feedbackScore: "",
  userComment: "",
}
```

Another approach here would be to have typed allow list keys _and_ values, but this is a more complex task without a properly typed graph (we'll just have to resort to casting or parsing via Zod to make it actually work currently).